ignite/engine/state.py

```diff
@@ -64,18 +79,29 @@ def __init__(self, **kwargs: Any) -> None:
         for k, v in kwargs.items():
             setattr(self, k, v)
 
-        self._update_attrs()
+        # If it's a standalone state with no engine provided, then we must update
+        # all the attributes and set all initial values to zeros,
+        # epoch and iteration included. We do this to keep BC compatibility,
+        # as we now don't keep iteration and epoch here in state,
+        # instead of that, we use engine._allowed_events_counts to keep track of them.
+        if self._engine is None:
+            self._update_attrs()
 
     def _update_attrs(self) -> None:
-        for value in self.event_to_attr.values():
-            if not hasattr(self, value):
-                setattr(self, value, 0)
+        for key in self.attr_to_events.keys():
+            if not hasattr(self, key):
+                setattr(self, key, 0)
 
     def get_event_attrib_value(self, event_name: Union[str, Events, CallableEventWithFilter]) -> int:
         """Get the value of Event attribute with given `event_name`."""
-        if event_name not in State.event_to_attr:
+        event_to_attr = {}
+        for attribute, events in self._attr_to_events.items():
+            for event in events:
+                event_to_attr[event] = attribute
+
+        if event_name not in event_to_attr:
             raise RuntimeError(f"Unknown event name '{event_name}'")
-        return getattr(self, State.event_to_attr[event_name])
+        return getattr(self, event_to_attr[event_name])  # type: ignore
```

Comment:
I think `get_event_attrib_value` can go to `EventsDrivenState` and be implemented like:
```python
    def get_event_attrib_value(self, event_name) -> int:
        if event_name not in self.engine._allowed_events_counts:
            raise RuntimeError(f"Unknown event name '{event_name}'")        
        return self.engine._allowed_events_counts[event_name]
```